pkg/capacity/manager.go

```diff
@@ -89,17 +81,62 @@ func hasPlacementRoles(placementRoles []string) spaceProvisionerConfigPredicate
 	}
 }
 
-func NewClusterManager(namespace string, cl runtimeclient.Client) *ClusterManager {
+func DefaultClusterManager(namespace string, cl runtimeclient.Client) *ClusterManager {
+	return NewClusterManager(namespace, cl, nil)
+}
+
+func NewClusterManager(namespace string, cl runtimeclient.Client, getSpaceCount SpaceCountGetter) *ClusterManager {
 	return &ClusterManager{
-		namespace: namespace,
-		client:    cl,
+		namespace:     namespace,
+		client:        cl,
+		getSpaceCount: getSpaceCount,
```

Comment:
I think that we discussed that it would make more sense having the changes of the capacity manager in a separate PR, that it would help us with promoting the changes in an isolated way.

Reply:
ah, true. It's been so long that I completely forgot about this.